### DISCUSSION TITLE
Update dimension ordering to match NetCDF-CF conventions

### DISCUSSION THREAD

**Discussion by tylere** [google/Xee#196] [Ideas]
Most Python libraries that plot raster images use a (y, x) or (lat, lon) ordering of dimensions. This is the convention used by [Matplotlib's pyplot with images](https://matplotlib.org/stable/tutorials/images.html), and many libraries that utilize it ([Xarray](https://docs.xarray.dev/en/stable/user-guide/plotting.html), [rasterio](https://rasterio.readthedocs.io/en/stable/topics/plotting.html)).

(lat, lon) ordering is also used for higher dimensional data that follows the netCDF-CF conventions, which are a standard for climate forecast data. Specifically:
> https://ferret.pmel.noaa.gov/Ferret/documentation/coards-netcdf-conventions
> **Order of dimensions:**
> If any or all of the dimensions of a variable have the interpretations of "date or time" (a.k.a. "T"), "height or depth"(a.k.a. "Z"), "latitude" (a.k.a. "Y"), or "longitude" (a.k.a. "X") then those dimensions should appear in the relative order T, then Z, then Y, then X in the CDL definition corresponding to the file.

By contrast Xee currently creates an Xarray dataset with [dimensions in the order (time, lon, lat)](https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L247) in which the lon and lat coordinates are swapped. This requires the user to process (transpose) the data before using Matplotlib or similar plotting libraries.

This convention difference may have contributed to several the the Xee issues that have been raised (#45, #171, #174). 

I think it may be worth changing Xee's coordinate ordering to (time, lat, lon) in order to match the convention used by netCDF-CF and Matplotlib. I realize this will be a breaking change for some users, but it will make the library easier to use for future users.

Thoughts?

**Comment by tylere** [comment]
The coordinate ordering logic appears in several places in Xee:
https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L245
https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L390
https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L570
https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L717
https://github.com/google/Xee/blob/5ce8db60d8cf78c4967ee21016d802add1d4d3cb/xee/ext.py#L724

**Reply by alxmrs** [reply]
This sounds like a great idea! The CF conventions do fit better with my expectations as an Xarray user. 

I think this is worth the breaking change. If I recall correctly, the order that's there now resulted from trying not to adulterate what was returned from EE. I'm not convinced that's the best precedent to support going forward.

**Comment by jdbcode** [comment]
A request for this change: #230

**Comment by tylere** [comment]
Implementation tracked with https://github.com/google/Xee/pull/275

**Comment by jdbcode** [comment]
The implementation is now available as prerelease 0.1.0 on PyPI. Please see this request for help with testing: https://github.com/google/Xee/discussions/303
